packages/edit-site/src/components/sidebar-edit-mode/template-panel/hooks.js

```diff
@@ -38,17 +41,17 @@ function preparePatterns( patterns, template, currentThemeStylesheet ) {
 	const filterOutDuplicatesByName = ( currentItem, index, items ) =>
 		index === items.findIndex( ( item ) => currentItem.name === item.name );
 
-	// Filter out core patterns.
-	const filterOutCorePatterns = ( pattern ) =>
-		! PATTERN_CORE_SOURCES.includes( pattern.source );
+	// Filter out core/directory patterns not included in theme.json.
+	const filterOutExcludedPatternSources = ( pattern ) =>
+		! EXCLUDED_PATTERN_SOURCES.includes( pattern.source );
 
 	// Filter only the patterns that are compatible with the current template.
 	const filterCompatiblePatterns = ( pattern ) =>
 		pattern.templateTypes?.includes( template.slug );
 
 	return patterns
 		.filter(
-			filterOutCorePatterns &&
+			filterOutExcludedPatternSources &&
 				filterOutDuplicatesByName &&
 				filterCompatiblePatterns
 		)
```

Comment:
It's unusual that the functions aren't called here ... Is this a bug? I think it should be:
```js
return patterns
		.filter( ( pattern) => 
			filterOutExcludedPatternSources( pattern ) &&
				filterOutDuplicatesByName( pattern ) &&
				filterCompatiblePatterns( pattern )
		)
```

Reply:
hmm, well spotted. In my testing of this only the last method is called. I will put up a separate PR for this as will be easier to review/test in isolation.

Reply:
[PR to update this here](https://github.com/WordPress/gutenberg/pull/55980)